### ISSUE TITLE
Need ability to programmatically create a SparkContext.

### ISSUE THREAD

**Issue by bpburns** [ibm-et/spark-kernel#159]
There should be mode where the kernel can be instantiated without creating a SparkContext.  This way the SparkContext can be created via code execution. 
/cc @lbustelo  
/cc @chipsenkbeil

**Comment by chipsenkbeil** [comment]
Things to consider:

1. Toggling this through command line and environment variable options

    ```bash
    sparkkernel --no-spark-context ...
    ```

    ```bash
    NO_SPARK_CONTEXT="true" sparkkernel ...
    ```

2. Adding the Spark Context created manually to be available for magics

    ```scala
    val sparkContext = new SparkContext(myConfig)
    kernel.registerSparkContext(sparkContext) // Puts in our dependency map so magics have it available
    ```

3. Avoiding issues where the created Spark Context variable is not transient, maybe by forcing creation through a magic that adds the variable to the interpreter

    ```scala
    // Cell 1
    %createSparkContext myConfig "mySparkContext"

    // Cell 2
    // @transient val mySparkContext = new SparkContext(myConfig)
    mySparkContext // Now available after running the magic, will also register the context for you
    ```

**Comment by lbustelo** [comment]
This is resolved by PR #191
